lib/util/DiUtil.js

```diff
@@ -59,3 +70,70 @@ export function hasEscalationEventDefinition(element) {
 export function hasCompensateEventDefinition(element) {
   return hasEventDefinition(element, 'bpmn:CompensateEventDefinition');
 }
+
+function hasCommonBoundaryIntermediateEventDefinition(element) {
+  return hasOneOfEventDefinitions(element, [
+    'bpmn:MessageEventDefinition',
+    'bpmn:TimerEventDefinition',
+    'bpmn:SignalEventDefinition',
+    'bpmn:ConditionalEventDefinition'
+  ]);
+}
+
+function hasOneOfEventDefinitions(element, eventDefinitions) {
+  return eventDefinitions.some(function(definition) {
+    return hasEventDefinition(element, definition);
+  });
+}
+
+/**
+ * Return the parent of the element with any of the given types.
+ *
+ * @param {djs.model.Base} element
+ * @param {String|Array<String>} anyType
+ *
+ * @return {djs.model.Base}
+ */
+export function getParent(element, anyType) {
+
+  if (typeof anyType === 'string') {
+    anyType = [ anyType ];
+  }
+
+  while ((element = element.parent)) {
+    if (isAny(element, anyType)) {
+      return element;
+    }
+  }
+
+  return null;
+}
+
+/**
+ * We treat IntermediateThrowEvents as boundary events during create,
+ * this must be reflected in the rules.
+ */
+export function isBoundaryCandidate(element) {
+  if (isBoundaryEvent(element)) {
+    return true;
+  }
+
+  if (is(element, 'bpmn:IntermediateThrowEvent') && hasNoEventDefinition(element)) {
+    return true;
+  }
+
+  return (
+    is(element, 'bpmn:IntermediateCatchEvent') &&
+    hasCommonBoundaryIntermediateEventDefinition(element)
+  );
+}
+
+export function isBoundaryEvent(element) {
```

Comment:
How is this related to the DI?

Reply:
It's not 👍. I think a better place is under [lib/util/ModelUtil.js](https://github.com/bpmn-io/bpmn-js/pull/1215/files/3a709634185db6e85631c7badb716a6831517cf9?file-filters%5B%5D=.js&hide-deleted-files=true#diff-9fb402b8a60ec6b0c96714019a5756b7).